bin/tools/install_tools.ml

```diff
@@ -0,0 +1,41 @@
+open! Import
+module Pkg_dev_tool = Dune_rules.Pkg_dev_tool
+
+let build_dev_tools dev_tools common =
+  let open Fiber.O in
+  let+ result =
+    let dev_tools_path =
+      List.map dev_tools ~f:(fun path -> Pkg_dev_tool.exe_path path |> Path.build)
+    in
+    Build_cmd.run_build_system ~common ~request:(fun _build_system ->
+      List.map dev_tools_path ~f:Action_builder.path |> Action_builder.all_unit)
+  in
+  match result with
+  | Error `Already_reported -> raise Dune_util.Report_error.Already_reported
```

Comment:
I wonder, shouldn't this be `Code_error.raise`?

Reply:
If there is an error, it is already reported during the build and `Code_error.raise` is about raising with a message.